### ISSUE TITLE
Support BigQuery Pipe Syntax

### ISSUE THREAD

**Issue by andres-lowrie** [sqlfluff/sqlfluff#6630]
### Description

Hello 👋 

BigQuery supports a new "pipe syntax"

https://cloud.google.com/blog/products/data-analytics/simplify-your-sql-with-pipe-syntax-in-bigquery-and-cloud-logging?e=48754805

and I'm wondering if adding support for it would make sense for this project. 

----

Currently, unsurprisingly, it fails to parse 

```sql
FROM myproject.mydataset.mytable
|> WHERE foo = 1
|> SUM(bar) AS sum_bar
```

```
sqlfluff --version
sqlfluff, version 2.3.5

sandbox sqlfluff lint --dialect bigquery foo.sql
== [foo.sql] FAIL                                                                                                                                                                                                                                                                                                         
    
L:   1 | P:   1 |  PRS | Line 1, Position 1: Found unparsable section: 'FROM
                       | myproject.mydataset.mytable\n|> WHER...'
WARNING: Parsing errors found and dialect is set to 'bigquery'. Have you configured your dialect correctly?
All Finished 📜 🎉! 
```

### Use case

I think a good first pass at this would be to just allow parsing to work. Then another step would be to add rules like "make sure the pipes are aligned" 

so 

```sql
FROM sometable |> expression |> expression
```

would become
```sql
FROM sometable
|> expression
|> expression
```



### Dialect

BigQuery

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by antoine-lefebvre** [comment]
As we are running SQL Fluff in your CI/CD pipeline to ensure our models are linted, as of today, we cannot use the BigQuery Pipe Syntax. Having this support would be tremendous!
